src/module.js

```diff
@@ -6,10 +6,15 @@ import rethrow from "./rethrow";
 import {variable_invalidation, variable_visibility} from "./runtime";
 import Variable, {TYPE_DUPLICATE, TYPE_IMPLICIT, TYPE_NORMAL, no_observer} from "./variable";
 
-export default function Module(runtime) {
+export default function Module(runtime, special = {}) {
```

Comment:
No need for this to have a default argument value if *runtime*.module already has it.

Reply:
(Though you may wish to change the other call to new Module to pass in the given special.)

Reply:
That caught a few cases of derive which should copy any specials into the derived module.